sdk/storage/storage-blob/review/generated/src/models/index.d.ts

```diff
@@ -0,0 +1,6713 @@
+/**
+ * signed identifier
+ */
+export interface SignedIdentifier {
+    /**
+     * a unique id
+     */
+    id: string;
+    accessPolicy: AccessPolicy;
+}
+/**
+ * The properties that enable an account to host a static website
+ */
+export interface StaticWebsite {
+    /**
+     * Indicates whether this account is hosting a static website
+     */
+    enabled: boolean;
+    /**
+     * The default name of the index page under each directory
+     */
+    indexDocument?: string;
+    /**
+     * The absolute path of the custom 404 page
+     */
+    errorDocument404Path?: string;
+}
+/**
+ * Storage Service Properties.
+ */
+export interface StorageServiceProperties {
+    logging?: Logging;
+    hourMetrics?: Metrics;
+    minuteMetrics?: Metrics;
+    /**
+     * The set of CORS rules.
+     */
+    cors?: CorsRule[];
+    /**
+     * The default version to use for requests to the Blob service if an incoming request's version
+     * is not specified. Possible values include version 2008-10-27 and all more recent versions
+     */
+    defaultServiceVersion?: string;
+    deleteRetentionPolicy?: RetentionPolicy;
+    staticWebsite?: StaticWebsite;
+}
+/**
+ * Stats for the storage service.
+ */
+export interface StorageServiceStats {
+    geoReplication?: GeoReplication;
+}
+/**
+ * Additional parameters for a set of operations.
+ */
+export interface LeaseAccessConditions {
+    /**
+     * If specified, the operation only succeeds if the resource's lease is active and matches this
+     * ID.
+     */
+    leaseId?: string;
+}
+/**
+ * Additional parameters for a set of operations.
+ */
+export interface ModifiedAccessConditions {
+    /**
+     * Specify this header value to operate only on a blob if it has been modified since the
+     * specified date/time.
+     */
+    ifModifiedSince?: Date;
+    /**
+     * Specify this header value to operate only on a blob if it has not been modified since the
+     * specified date/time.
+     */
+    ifUnmodifiedSince?: Date;
+    /**
+     * Specify an ETag value to operate only on blobs with a matching value.
+     */
+    ifMatch?: string;
+    /**
+     * Specify an ETag value to operate only on blobs without a matching value.
+     */
+    ifNoneMatch?: string;
+}
+/**
+ * Additional parameters for a set of operations, such as: Directory_create, Directory_rename,
+ * Blob_rename.
+ */
+export interface DirectoryHttpHeaders {
+    /**
+     * Cache control for given resource
+     */
+    cacheControl?: string;
+    /**
+     * Content type for given resource
+     */
+    contentType?: string;
+    /**
+     * Content encoding for given resource
+     */
+    contentEncoding?: string;
+    /**
+     * Content language for given resource
+     */
+    contentLanguage?: string;
+    /**
+     * Content disposition for given resource
+     */
+    contentDisposition?: string;
+}
+/**
+ * Additional parameters for a set of operations.
+ */
+export interface SourceModifiedAccessConditions {
```

Comment:
Do we need this type; it is identical to ModifiedAccessConditions except for "source".

Reply:
I guess not. Our library uses `ModifiedAccessConditions`